pkg/build/api/types.go

```diff
@@ -328,6 +328,9 @@ type GitBuildSource struct {
 
 	// HTTPSProxy is a proxy used to reach the git repository over https
 	HTTPSProxy *string
+
+	// URLCheckTimeoutSeconds is the timeout in seconds to use when checking connectivity to the git server
+	URLCheckTimeoutSeconds *int
```

Comment:
@deads2k I don't know we can apply it across the board to all repository types. This is a very git-specific check (git ls-remote)